src/lib/getAllPackages.ts

```diff
@@ -109,6 +183,57 @@ async function getWorkspacePackageInfos(
   return [selectedWorkspacePackageInfos, allWorkspacePackageNames]
 }
 
+/**
+ * Gets pnpm catalog package info from pnpm-workspace.yaml.
+ *
+ * @param options the application options
+ * @param rootPackageFile the root package file path
+ * @returns PackageInfo for catalog dependencies or null if no catalogs exist
+ */
+async function getCatalogPackageInfo(options: Options, rootPackageFile: string): Promise<PackageInfo | null> {
+  const { pkgPath } = await findPackage({ ...options, packageFile: rootPackageFile, loglevel: 'silent' })
```

Comment:
Is the `pkgPath` already available? It would be nice if we didn't have to call `findPackage` again, since it involves multiple round trips to disk.